src/components/Episodes/context.jsx

```diff
@@ -1,23 +1,58 @@
-import React, { createContext } from "react";
-import { data } from "./";
+import React, { createContext, useReducer } from "react";
+import { 
+	reducer, 
+	data
+} from "./";
 
-const EpisodesContext = createContext();
 
-const episodes = data.length;
+// EPISODES CONTEXT
+// ----------------------------
+const EpisodesContext = createContext();
 
 const initialState = {
 	activeIndex: 0,
-	episodeCount: episodes
-};
+	overlayActive: false,
+	episodeCount: data.length
+}
 
 function EpisodesProvider(props){
 	const { children } = props;
 
+	const episodeCount = data.length;
+	const [state, dispatch] = useReducer(reducer, initialState);
+
 	return (
-		<EpisodesContext.Provider value={ initialState }>
+		<EpisodesContext.Provider value={ {state, dispatch} }>
 			{ children }
 		</EpisodesContext.Provider>
 	);
-}
+}// EpisodesProvider
+
+
+// TRAILER CONTEXT
+// ------------------------------
+const TrailersContext = createContext();
```

Comment:
I think this context should be in `Episode/context.jsx` as it's going to be part of the individual episode render, if ya get me?